proposals/2024-12-14_update-created-timestamps-for-openmetrics-text.md

```diff
@@ -0,0 +1,151 @@
+This change is not backwards compatible and would break existing parsers that expect the `_created` line. OpenMetrics 1.x parsers that support `_created` lines would not be able to parse the new syntax. This would require a new major version of the OpenMetrics specification to be released, i.e, OpenMetrics 2.0. Any client libraries or tools that expose OpenMetrics text would also need to be updated to support the new syntax.
+
+## Alternatives
+
+### Do nothing
+
+Continue using the OM text format in its current state and further optimize CPU usage through PRs that build on [github.com/prometheus/prometheus/issues/14823](https://github.com/prometheus/prometheus/issues/14823). This is desirable if we wish to keep backwards compatibility and avoids breaking changes but we would have to live with an inefficient solution.
+
+### Storing CTs Using a `# HELP`-Like Syntax
+
+In addition to the `TYPE`, `UNIT`, and `HELP` fields, we can introduce a `# CREATED` line for metrics that have an associated creation timestamp (CT). This approach allows us to quickly determine whether a CT exists for a given metric, eliminating the need for a more time-consuming search process. By parsing the `# CREATED` line, we can associate it with a specific hash corresponding to the metric's label set, thereby mapping each CT to the correct metric.
+
+However, this method still involves the overhead of storing potentially multiple CTs in memory until we encounter the relevant metric line. Furthermore the `CREATED` line itself might look somewhat convoluted compared to `TYPE` , `UNIT` , and `HELP`  which are very human readable. 
+
+An example of how this might look:
+
+```
+# HELP foo Counter with and without labels to certify CT is parsed for both cases
+# TYPE foo counter
+# CREATED 1520872607.123; {a="b"} 1520872607.123; {le="c"} 1520872621.123
+foo_total 17.0 1520879607.789 # {id="counter-test"} 5
+foo_total{a="b"} 17.0 1520879607.789 # {id="counter-test"} 5
+foo_total{le="c"} 21.0
+foo_total{le="1"} 10.0
+```
+
+When the same MetricFamily has multiple label sets with their own CTs we'd have to place all the timestamps in a single line with the additional labels and separate them with delimiters. This does mitigate the increased verbosity described in the [Summaries and Histograms section](#summaries-and-histograms) but it still requires storing all the timestamps in memory for the lifetime of the MetricFamily. Even if we store the CT only for the first label set in a MetricFamily, we only need to keep a single timestamp in memory at a time for each label set in a summary or histogram until the next label set is encountered (assuming metrics with the same label sets are grouped together).
+
+## Action Plan
+
+In no particular order:
+
+* [ ] Update the OpenMetrics specification to include the new syntax for Created Timestamps.
+* [ ] Update Prometheus's OpenMetrics text parser to support the new syntax.
+* [ ] Update tooling and libraries that expose OpenMetrics text. 
```

Comment:
I think we need an order, actually!

First update libraries to expose the new format, then there's data to test the new parser and lastly, if the change show performance improvement, we can commit the idea in OM 2.0